### ISSUE TITLE
StringTransport's disconnecting instance variable is initialized to zero, not False, contrary to the documentation

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4251]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)| @jesstess reported|
|-|-|
|Trac ID|trac#4251|
|Type|defect|
|Created|2010-02-02 17:11:33Z|
|Branch|https://github.com/twisted/twisted/tree/stringtransport-disconnecting-4251|

Attachments:

* [proto_helpers_false.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/542/54236503778c01a5da6c782c70c2256786be5950/c01a713d7b2c909839bde0513ab4b46fe414617b.patch) (385 bytes) - added by jesstess on 2010-02-07 18:56:03Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4251 4251
type__defect defect
reporter__jesstess jesstess
priority__normal normal
milestone__ 
branch__branches_stringtransport_disconnecting_4251 branches/stringtransport-disconnecting-4251
branch_author__jesstess jesstess
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1265130693000000 1265130693000000
changetime__1265639620000000 1265639620000000
version__None None
owner__ 
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_1"></a>|@jesstess removed owner|
|-|-|

Since there's no functional change except to the ```str()``` and ```repr()``` of ```disconnecting```, can I get away with no unit test and just get a review and apply this on trunk?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @jesstess|
|-|-|

It looks like there's current no test coverage for the `disconnecting` attribute at all.  How about adding one that verifies it starts as a false value and becomes true when `loseConnection` is called.

Presumably the test would pass before the proposed change, so it's not like this is going to prove that `False` is vastly superior to `0`, but as long as someone is looking at this code, we may as well improve test coverage a bit. :)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_3"></a>|@jesstess commented|
|-|-|

(In [28123]) Branching to 'stringtransport-disconnecting-4251'

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_4"></a>|@jesstess commented|
|-|-|

(In [28124]) Add a docstring, unit test, and topfile.

refs [#4251](https://github.com/twisted/twisted/issues/4251)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_6"></a>|@mwhudson commented|
|-|-|

I think using assertIdentical to check that disconnecting really is False, then True would be justified here.

Please merge after making that change.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_7"></a>|@jesstess commented|
|-|-|

(In [28127]) Use assertFalse and assertTrue.

refs [#4251](https://github.com/twisted/twisted/issues/4251)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_8"></a>|@jesstess set status to `closed`|
|-|-|

(In [28128]) Merge stringtransport-disconnecting-4251

Author: jesstess
Reviewers: mwh
Fixes: [#4251](https://github.com/twisted/twisted/issues/4251)

Initialize the disconnecting instance variable of StringTransport to
False instead of 0.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
